app/src/main/java/org/p2p/wallet/infrastructure/network/provider/TokenKeyProvider.kt

```
@@ -19,20 +19,24 @@ class TokenKeyProvider(
 
     private val listeners = mutableListOf<TokenKeyProviderListener>()
 
-    var publicKey: String = getPublicKeyFromStorage()
+    var publicKey: String = emptyString()
+        get() = getPublicKeyFromStorage()
         set(value) {
             field = value
             savePublicKeyToStorage(value)
             Timber.tag(TAG).i("updating user public key: $value")
         }
 
-    private fun getPublicKeyFromStorage(): String {
-        return runBlocking {
+    private fun getPublicKeyFromStorage(): String = runBlocking {
+        try {
             val base58String = secureStorage.getString(Key.KEY_PUBLIC_KEY).orEmpty()
             Base58Utils.decodeToString(base58String)
+        } catch (e: Throwable) {
+            Timber.tag(TAG).e(e)
+            throw e
```

throw anyway, we need to fail fast